Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -172,13 +172,13 @@ internal struct _BSONDecodingStorage {
     fileprivate var count: Int { return self.containers.count }
 
     /// The container at the top of the stack.
-    internal var topContainer: BSONValue? {
+    internal var topContainer: BSONValue {
```

Comment:
In light of https://github.com/mongodb/mongo-swift-driver/pull/145/files#r235482868, I just want to confirm that the precondition here ensures that we never access `topContainer` when the `containers` array is empty. Therefore, this is always returning an actual BSONValue, which is why we can force-unwrap in the `return` statement.

Reply:
Yep, that's exactly what the `precondition` exists here for. In general, the only case where a force-unwrap is tolerable in good Swift is when its backed by a `precondition` or its something you can reasonably assume to be true in a test, so this seems fine in terms of both correctness and code quality.